toqito/matrices/tests/test_operator_sinkhorn.py

```diff
@@ -0,0 +1,182 @@
+"""Tests for operator_sinkhorn."""
+
+import warnings
+
+import numpy as np
+import pytest
+
+from toqito.channels.partial_trace import partial_trace
+from toqito.matrices.operator_sinkhorn import operator_sinkhorn
+from toqito.rand import random_density_matrix
+from toqito.states import bell
+
+
+def test_operator_sinkhorn_unitary_invariance():
+    """Test invariance of Operator Sinkhorn on swapping subsystems."""
+    rho = random_density_matrix(4)
+
+    U = np.kron(np.eye(2), np.array([[0, 1], [1, 0]]))  # Swap subsystem
+    rho_new = U @ rho @ U.conj().T
+    sigma_new, F_new = operator_sinkhorn(rho_new)
+    sigma_old, F_old = operator_sinkhorn(rho)
+    np.testing.assert_allclose(sigma_new, U @ sigma_old @ U.conj().T)
+
+
+def test_operator_sinkhorn_trace_property():
+    """Test the trace property of operator Sinkhorn when a valid inputs are passed."""
+    rho = random_density_matrix(9)  # 9-dimensional density matrix
+
+    # The dimension `3` divides evenly into `9`, so no error should occur
+    sigma, F = operator_sinkhorn(rho, dim=[3])
+
+    # Check that sigma is a valid density matrix with trace equal to 1
+    np.testing.assert_almost_equal(np.trace(sigma), 1)
+
+
+# Test partial traces for bipartites and tripartites systems
+
+"""Test operator Sinkhorn partial trace on a bipartite system."""
+# Generate a random density matrix for a 3 qubit x 3qubit system (9-dimensional).
+rho_bi = random_density_matrix(9)
+dim_bi = [3, 3]
+expected_identity_bi = np.eye(3) * (1 / 3)
+subsystem_bi = 0  # first subsystem only
+
+"""Test operator Sinkhorn partial trace on a tripartite system."""
+rho_tri = random_density_matrix(8)
+dim_tri = [2, 2, 2]
+expected_identity_tri = np.eye(2) * (1 / 2)
+subsystem_tri = [0, 2]
+
+params0 = [
+    ([rho_bi, dim_bi], [expected_identity_bi, subsystem_bi]),
+    ([rho_tri, dim_tri], [expected_identity_tri, subsystem_tri]),
+]
+
+
+@pytest.mark.parametrize("test_input,expected", params0)
+def test_operator_sinkhorn_partial_trace(test_input, expected):
+    """Test partial traces for bipartites and tripartites systems."""
+    rho1, dim1 = test_input
+    sigma, F = operator_sinkhorn(rho=rho1, dim=dim1)
+
+    # Expected partial trace should be proportional to identity matrix.
+    expected_identity, target_subsys = expected
+
+    # Partial trace on the first subsystem.
+    pt = partial_trace(sigma, target_subsys, dim=dim1)
+    pt_rounded = np.around(pt, decimals=2)
+
+    # Check that partial trace matches the expected identity.
+    np.testing.assert_allclose(pt_rounded, expected_identity, rtol=1e-2)
+
+
+# Test Error handling in different scenarios
+
+"""Test operator Sinkhorn with a singular matrix that triggers LinAlgError."""
```

Comment:
These test cases should be within the paramtereize call, not defined outside of it. Refer to something that uses `pytest.paramterize` within the project (one such example, out of many, could be `test_is_pseudo_hermitian.py` which uses the convention we want to adopt here.